blivet/devicetree.py

```diff
@@ -203,6 +203,12 @@ def _remove_device(self, dev, force=None, modparent=True):
                        device.disk == dev.disk:
                         device.update_name()
 
+        if isinstance(dev, LVMLogicalVolumeDevice) and dev.from_lvs:
+            # if removing an LV constructed from other LVs, we need to put the
+            # LVs it's supposed to be constructed from back into the device tree
+            for lv in dev.from_lvs:
+                self._add_device(lv, new=False)
```

Comment:
When would you remove such a device other than in the context of cancelling an action? The only example I can think of is trying to clear out a devicetree, in which case we should certainly not be quietly adding devices back into the tree.